File: python/sorting_algos/SelectionSort/selection_sort.py
def selection_sort(arr):
    n = len(arr)
    # Traverse through all elements of the array
    for i in range(n):
        # Find the minimum element in the remaining unsorted array
        min_idx = i
        for j in range(i+1, n):
            if arr[j] < arr[min_idx]:
                min_idx = j
        # Swap the found minimum element with the first element
        arr[i], arr[min_idx] = arr[min_idx], arr[i]

# Test the selection_sort function
arr = [30, 40, 20, 50, 10]
print("Original array:", arr)
selection_sort(arr)
print("Sorted array:", arr)

File: src/java/sorting_algos/SelectionSort.java
public class SelectionSort {
    public static void selectionSort(int[] arr) {
        int n = arr.length;
        // One by one move the boundary of the unsorted subarray
        for (int i = 0; i < n-1; i++) {
            // Find the minimum element in unsorted array
            int minIndex = i;
            for (int j = i+1; j < n; j++) {
                if (arr[j] < arr[minIndex]) {
                    minIndex = j;
                }
            }
            // Swap the found minimum element with the first element
            int temp = arr[minIndex];
            arr[minIndex] = arr[i];
            arr[i] = temp;
        }
    }

    // Utility method to print the array
    public static void printArray(int[] arr) {
        for (int value : arr) {
            System.out.print(value + " ");
        }
        System.out.println();
    }

    // Main method to test the SelectionSort class
    public static void main(String[] args) {
        int[] arr = {20,30,50,40,10};
        System.out.println("Original array:");
        printArray(arr);
        selectionSort(arr);
        System.out.println("Sorted array:");
        printArray(arr);
    }
}